### PR TITLE
rev up Kryo Serializers + add a few for collections

### PR DESCRIPTION
**Pull request by akiezun**
we've been using an ancient one.

@lbergelson please review

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/engine/spark/GATKRegistrator.java [hunk 1]
```diff
@@ -34,9 +34,18 @@ public void registerClasses(Kryo kryo) {
...
         kryo.register(Collections.unmodifiableList(Collections.EMPTY_LIST).getClass(), new UnmodifiableCollectionsSerializer());
 
+        kryo.register( Arrays.asList( "" ).getClass(), new ArraysAsListSerializer() );
+        kryo.register( Collections.EMPTY_LIST.getClass(), new CollectionsEmptyListSerializer() );
+        kryo.register( Collections.EMPTY_MAP.getClass(), new CollectionsEmptyMapSerializer() );
+        kryo.register( Collections.EMPTY_SET.getClass(), new CollectionsEmptySetSerializer() );
+        kryo.register( Collections.singletonList( "" ).getClass(), new CollectionsSingletonListSerializer() );
+        kryo.register( Collections.singleton( "" ).getClass(), new CollectionsSingletonSetSerializer() );
+        kryo.register( Collections.singletonMap( "", "" ).getClass(), new CollectionsSingletonMapSerializer() );
+        UnmodifiableCollectionsSerializer.registerSerializers( kryo );
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/engine/spark/GATKRegistrator.java]: Does this make registering `collections.unmodifiableList(Collections.EMPTY_LIST) .getClass()` and friends redundant?

**Comment by lbergelson** [comment]
@akiezun Kryo is a provided library, so reving it here may not make a difference if our cluster isn't upgraded as well I believe.

**Comment by lbergelson** [comment]
Looks good modulo the possibly redundant registration I asked about.

**Comment by akiezun** [comment]
let's close this. i dont have a proof that it improves anything

**Pull request closed without merging**